### PR TITLE
[UI Toolkit] Add cart functionality to base Theme Controller

### PR DESCRIPTION
**Pull request by JakeCataford**
Adds some helper methods for managing the cart with the theme
controller.

### CODE REVIEW

File: Assets/Shopify/UIToolkit/Theme Controllers/SingleProductThemeController.cs [hunk 1]
```diff
@@ -8,7 +8,16 @@
...
-        public ISingleProductTheme Theme;
+        public new ISingleProductTheme Theme { 
+            get { 
+                return base.Theme as ISingleProductTheme; 
+            }
+
+            set {
+                base.Theme = value;
+            } 
+        }
```
Comment (JakeCataford) [inline Assets/Shopify/UIToolkit/Theme Controllers/SingleProductThemeController.cs]: this is a neat trick for hiding the theme from the base controller and always getting the more specific interface

Comment (sleroux) [reply Assets/Shopify/UIToolkit/Theme Controllers/SingleProductThemeController.cs]: Neat trick! I had no idea you could use the `new` keyword like that.

File: Assets/Shopify/UIToolkit/Theme Controllers/ThemeControllerBase.cs [hunk 2]
```diff
@@ -48,5 +54,64 @@ public abstract class ThemeControllerBase : MonoBehaviour {
 
         public abstract void OnShow();
         public abstract void OnHide();
+        /// <summary>
+        /// The active cart that the controller is using.
+        /// </summary>
+        /// <returns>The Cart</returns>
+        public Cart Cart { get { return Client.Cart(); } }
```
Comment (sleroux) [inline Assets/Shopify/UIToolkit/Theme Controllers/ThemeControllerBase.cs]: Nit: I would throw this stuff on a new line just to keep it in line with other getters we have in the SDK

File: Assets/Shopify/UIToolkit/Theme Controllers/ThemeControllerBase.cs [hunk 3]
```diff
@@ -13,6 +17,8 @@ public abstract class ThemeControllerBase : MonoBehaviour {
         /// </summary>
         public string AccessToken;
 
+
```
Comment (richardmonette) [inline Assets/Shopify/UIToolkit/Theme Controllers/ThemeControllerBase.cs]: Nit: extra spaces

**Pull request merged**